packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -16,7 +16,7 @@ import 'debug.dart';
 /// During painting, the render tree generates a tree of composited layers that
 /// are uploaded into the engine and displayed by the compositor. This class is
 /// the base class for all composited layers.
-abstract class Layer {
+abstract class Layer extends TreeDiagnostics {
```

Comment:
I think I'd prefer "extends Object with TreeDiagnostics". Layer is not is-a TreeDiagnostics. Consider the case of a class that inherits from the AbstractNode class and wants this.